py3status/modules/sysdata.py

```diff
@@ -267,13 +264,36 @@ def _get_cputemp(self, zone, unit):
         m = re.search("(Core 0|CPU Temp).+\+(.+).+\(.+", sensors)
         if m:
             cpu_temp = float(m.groups()[1].strip()[:-2])
-            if unit == "K":
-                cpu_temp += 273.15
         else:
             cpu_temp = "?"
 
         return cpu_temp
 
+    def _get_cputemp(self, zone, unit):
+        # if unit not in [u"°F", u"°C", "K"]:
+        #     return "unknown unit"
```

Comment:
uncomment 😛

Reply:
Ah. I wasn't sure about this block, because the unit is also checked in `post_config_hook()`. Do we want to fail this way?

Reply:
Hmm you are right. I'm a bit surprized that we aren't raising an error in this case (not in `post_config_hook` either), but instead making the module return some string... Could you test what happens with this block commented when you use unknown unit, what do you see as a user? If it's clear for user that unit is wrong, I'm ok removing this commented block 👍

Reply:
Here's the relevant section from my config:

```
		sysdata tdie {
  		    zone = "/sys/class/hwmon/hwmon1/temp1_input"
			cache_timeout = 3
			thresholds = [(0, ""), (55, "degraded"), (65, "bad")]
			format =  '[\?color=temp '
			format += 'T_die: {cpu_temp:.0f}{temp_unit}'
			format += ']'
                        temp_unit = 'foo'
		}
```

With the block uncommented: `T_die unknown unitunknown unit`
With the block commented out: `T_die 30unknown unit`

Reply:
Awesome, let's remove this then 👍